### ISSUE TITLE
Problems with Shaarli v. 0.12.1

### ISSUE THREAD

**Issue by dajare** [kalvn/Shaarli-Material#119]
I've been using Material for a long time now with a particular Shaarli instance (beginning around v. 0.9, I think). I've been keeping things up-to-date, though, and today upgraded my Shaarli's to 0.12.1 (from 0.11.1). With the default template, everything is fine. With Material (latest!), however, I get this error message:

```
An unexpected error happened, and the error template could not be displayed.

Template error not found!
/home/mysite/public_html/shaarmarks/inc/rain.tpl.class.php268
#0 /home/mysite/public_html/shaarmarks/inc/rain.tpl.class.php(164): RainTPL->check_template('error')
#1 /home/mysite/public_html/shaarmarks/application/render/PageBuilder.php(253): RainTPL->draw('error', true)
#2 /home/mysite/public_html/shaarmarks/application/front/controller/visitor/ShaarliVisitorController.php(66): Shaarli\Render\PageBuilder->render('error', '/shaarmarks')
#3 /home/mysite/public_html/shaarmarks/application/front/controller/visitor/ErrorController.php(45): Shaarli\Front\Controller\Visitor\ShaarliVisitorController->render('error')
#4 [internal function]: Shaarli\Front\Controller\Visitor\ErrorController->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Error))
#5 /home/mysite/public_html/shaarmarks/vendor/slim/slim/Slim/App.php(728): call_user_func_array(Object(Shaarli\Front\Controller\Visitor\ErrorController), Array)
#6 /home/mysite/public_html/shaarmarks/vendor/slim/slim/Slim/App.php(396): Slim\App->handlePhpError(Object(Error), Object(Slim\Http\Request), Object(Slim\Http\Response))
#7 /home/mysite/public_html/shaarmarks/vendor/slim/slim/Slim/App.php(297): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))
#8 /home/mysite/public_html/shaarmarks/index.php(176): Slim\App->run(true)
#9 {main}
```

I have read through the "Compatibility with Shaarli v0.12.0-beta" #115 thread, and I'm none the wiser. Grateful for any help with this — I miss Material! (**Update** 2020-11-20 : I missed Material so much that I rolled that instance back to 0.11.1! So normal service is resumed, although it would be great to have this running with 0.12.1, too.)

**Comment by stoebern** [comment]
Exactly the same situation here. Thank you for any hint as where to go now.
Salutations

**Comment by dajare** [comment]
> Exactly the same situation here, even the same error message. Thank you for any hint as where to go now.
> Salutations

I rolled back to 0.11.1 Shaarli on the instance that I use with Material. Hopefully we can work out (or Kalvn can! 😬) how to get this working with 0.12.1.

**Comment by kalvn** [comment]
I didn't have enough time to start working on v0.12.1 (which seems to be massive) compatibility but I'll try to do that soon!

**Comment by dajare** [comment]
> I didn't have enough time to start working on v0.12.1 (which seems to be massive) compatibility but I'll try to do that soon!

Grateful for that! I've actually had a look a few times at trying to create a "minimalist" theme for Shaarli (something I've done with my CMS of choice quite few times over the years). It looks anything but easy with Shaarli, and all the trickier if a point-upgrade brings so many changes! So thanks for keeping Material up-to-date.

**Comment by immanuelfodor** [comment]
Many projects treat the 0th major release as unstable where even point releases can break things, and maintainers don't guarantee compatibility with earlier releases, so while this is being annoying, this is how the world works. Shaarli is about to get its 1st major release in some time, then point releases shouldn't break so many things: https://github.com/shaarli/Shaarli/issues/1640

**Comment by kalvn** [comment]
I just published v0.12.0 https://github.com/kalvn/Shaarli-Material/releases/tag/v0.12.0 which should solve that problem.

Working on v0.12.1 now :)

**Comment by stoebern** [comment]
> I just published v0.12.0 https://github.com/kalvn/Shaarli-Material/releases/tag/v0.12.0

Thank you very much. I'm only in this situation because I like your work so much. So thank you for sharing it!

**Comment by pascollin** [comment]
This theme is great ! But, when using it with shaarli v0.12.1, the red "plus" button is missing (the button used to add new links)

**Comment by immanuelfodor** [comment]
You can follow this issue here @pascollin : https://github.com/kalvn/Shaarli-Material/issues/117

**Comment by immanuelfodor** [comment]
I've rebuilt my Shaarli using the `support-v012` branch, and now all my previously observed issues are gone. There were some unnoticed changes on that branch in November, I suggest you all to try it :) Happy new year to everyone! :tada:

**Comment by dajare** [comment]
> I've rebuilt my Shaarli using the `support-v012` branch,...

Thanks for the pointer, @immanuelfodor — but I can't [identify any branch with that name](https://github.com/shaarli/Shaarli/branches/all). Has it been erased? or is the name not quite right? or am I looking in the wrong place?

> Happy new year to everyone! 🎉

Ditto! 👍

**Comment by kalvn** [comment]
Fixed in [v0.12.1](https://github.com/kalvn/Shaarli-Material/releases/tag/v0.12.1).

**Comment by vrykolas654** [comment]
FYI: I had similar symptoms (but different cause). My problem was due to folder/file permissions on the /data and /cache folders and subfolders.
When I reinstalled Synology DSM (this is where docker is installed and Shaarli is hosted)... I forgot about making the /data and /cache folders writeable to "users".

To fix MY problem: I stopped the container, stopped inheritence of permissions for /shaarli folder in docker, then made /data and /cache writeable to 'users'. Then I restarted the container and it started up with no problems!

Hopefully this anecdote/scenario can help someone else with a similar (Synology-related) problem in the future.
